src/pages/AuthPage.tsx

```diff
@@ -0,0 +1,320 @@
+  const [error, setError] = useState("")
+  const navigate = useNavigate()
+  const authError = useAppSelector((state) => state.search.error)
+
+  const [isFormValid, setIsFormValid] = useState(false)
+  const location = useLocation()
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      if (user) {
+        if (location.state) {
+          const { pathname, search } = location.state.from as any
+
+          navigate(`${pathname}${search}`)
+        } else {
+          dispatch(setCurrentUser(user.email!))
+          navigate("/search")
+        }
+      } else {
+        console.log("no user")
+      }
+    })
+
+    return () => {
+      unsubscribe()
+    }
+  }, [auth.currentUser])
+
+  useEffect(() => {
+    if (authError) {
+      setError(authError)
+    }
+  }, [authError])
+
+  useEffect(() => {
+    checkFormValidity()
+  }, [isLogin, email, password, passwordConfirm])
+
+  const checkFormValidity = () => {
+    if (isLogin) {
+      setIsFormValid(email !== "" && password !== "")
+    } else {
+      setIsFormValid(
+        email !== "" && password !== "" && password === passwordConfirm,
+      )
+    }
+  }
+
+  const handleSignIn = async (e: any) => {
+    e.preventDefault()
+    setError("")
+
+    if (!validateEmail(email)) {
+      setError("Please enter a valid email address")
+
+      return
+    }
+
+    if (!validatePassword(password)) {
+      setError("Please enter a valid password")
+
+      return
+    }
+
+    try {
+      await dispatch(signIn({ email, password })).then(() => {
+        if (location.state) {
+          const { pathname, search } = location.state.from as any
+
+          navigate(`${pathname}${search}`)
+        } else {
+          dispatch(setCurrentUser(auth.currentUser!.email!))
+          navigate("/search")
+        }
+      })
+    } catch (error_: any) {
+      setError(error_.message)
+    }
+  }
+
+  const handleSignUp = async (e: any) => {
+    e.preventDefault()
+    setError("")
+
+    if (!validateEmail(email)) {
+      setError("Please enter a valid email address")
+
+      return
+    }
+
+    if (!validatePassword(password)) {
+      setError(
+        "Please enter a valid password (min. 6 characters and includes at least one lowercase letter, one uppercase letter, and one number)",
+      )
+
+      return
+    }
+
+    if (password !== passwordConfirm) {
+      setError("Passwords do not match")
+
+      return
+    }
+
+    try {
+      await dispatch(signUp({ email, password }))
+    } catch (error_: any) {
+      setError(error_.message)
+    }
+  }
+
+  const handleEmailChange = (e: any) => {
+    setEmail(e.target.value)
+  }
+
+  const handlePasswordChange = (e: any) => {
+    setPassword(e.target.value)
+  }
+
+  const handlePasswordConfirmChange = (e: any) => {
+    setPasswordConfirm(e.target.value)
+  }
+
+  const validateEmail = (email: string) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
```

Comment:
```suggestion
    const emailRegex = /^[^\s@]+@[^\s@]+$/
```
it covers dots in the middle. And dot is not mandatory
https://david-gilbertson.medium.com/the-100-correct-way-to-validate-email-addresses-7c4818f24643